### PR TITLE
remove deprecated cli arguments from MO

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
 - removed deprecated cli arguments from MO except `data_type`, `disable_nhwc_to_nchw` and `tensorflow_use_custom_operations_config` since they are actively used in e2e
 - removed `ConcatOptimizations` since it was activated only when `enable_concat_optimization` arg is set and no other places where it was called
 - updated documents accordingly

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* n/a  Transformation generates reshape-able IR
* n/a  Transformation preserves original framework node names
* n/a  Transformation preserves tensor names

Validation:
* [x]  Unit tests
* n/a  Framework operation tests
* n/a  Transformation tests

Documentation:
*  n/a  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model: updated guides after deprecated argument are removed
* [x]  User guide update: updated guides after deprecated argument are removed

### Tickets:
 - xxx-99434, xxx-99259

### CODE REVIEW

**Comment by pavel-esir** [comment]
for some reason tmp irs cleanup is failing

File: tools/legacy/benchmark_app/README.md [hunk 1]
```diff
@@ -142,7 +142,7 @@ This section provides step-by-step instructions on how to run the Benchmark Tool
    omz_downloader --name googlenet-v1 -o <models_dir>
 2. Convert the model to the Inference Engine IR format. Run the Model Optimizer using the `mo` command with the path to the model, model format (which must be FP32 for CPU and FPG) and output directory to generate the IR files:
    ```sh
-   mo --input_model <models_dir>/public/googlenet-v1/googlenet-v1.caffemodel --data_type FP32 --output_dir <ir_dir>
+   mo --input_model <models_dir>/public/googlenet-v1/googlenet-v1.caffemodel --output_dir <ir_dir>
```
Comment (rkazants) [inline tools/legacy/benchmark_app/README.md:145]:
```suggestion
   mo --input_model <models_dir>/public/googlenet-v1/googlenet-v1.caffemodel
```

Comment (rkazants) [reply tools/legacy/benchmark_app/README.md:145]: output_dir looks ugly in the user guide.

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 2]
```diff
@@ -709,25 +709,6 @@ def transform_param_to_str(value):
...
-        setattr(namespace, self.dest, True)
-
-
-class DeprecatedOptionCommon(argparse.Action):
-    def __call__(self, parser, args, values, option_string):
-       dep_msg = "Use of deprecated cli option {} detected. Option use in the following releases will be fatal. ".format(option_string)
-       log.error(dep_msg, extra={'is_warning': True})
-       setattr(args, self.dest, values)
-
-
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:730]: Please do not remove if because we can deprecate future options. Just remove mention of `--finegrain_fusing`, etc.

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 3]
```diff
@@ -817,16 +798,6 @@ def __call__(self, parser, namespace, values, option_string=None):
...
-class DeprecatedCanonicalizePathCheckExistenceAction(CanonicalizePathCheckExistenceAction):
-    def __call__(self, parser, namespace, values, option_string=None):
-        dep_msg = "Use of deprecated cli option {} detected. Option use in the following releases will be fatal. ".format(
-            option_string)
-        if 'tensorflow_use_custom_operations_config' in option_string:
-            dep_msg += 'Please use --transformations_config cli option instead'
-        log.error(dep_msg, extra={'is_warning': True})
-        super().__call__(parser, namespace, values, option_string)
-
-
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:829]: the same comment

Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:829]: done

File: tools/mo/openvino/tools/mo/utils/cli_parser.py (hunk 2 above)
Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:730]: done

File: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md [hunk 4]
```diff
@@ -256,8 +256,6 @@ There are several middle transformations responsible for changing model layout f
 
 This layout change is disabled automatically if the model does not have operations that OpenVINO&trade needs to execute in the NCHW layout, for example, Convolutions in NHWC layout.
 
-It is still possible to force Model Optimizer to do a layout change, using `--disable_nhwc_to_nchw` command-line parameter, although it is not advised.
-
 
 Layout change is a complex problem and will be addressed here very briefly. For more details on how it works, refer to the source code of the transformations mentioned in the below summary of the process:
```
[the remark this answers is not in the document]
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md]: applied this suggestion

**Comment by pavel-esir** [comment]
> for some reason tmp irs cleanup is failing

fixed that. issue was caused because of relying of deprecated `data_type` in unit-tests

**Comment by pavel-esir** [comment]
E2E tests are failing. Need to remove `–data_type` from there as well. We will not get green pre-commit until E2E will be updated

**Comment by rkazants** [comment]
> E2E tests are failing. Need to remove `–data_type` from there as well. We will not get green pre-commit until E2E will be updated

Did you recover `data_type` option? Please do it asap.

File: tools/mo/openvino/tools/mo/middle/ConcatOptimization.py [hunk 5]
```diff
@@ -1,116 +0,0 @@
...
-            # re-connect input ports sequentially and remove all not used
-            port_idx_to_connect = 0
-            for port_idx in range(max_port_index + 1):
-                if concat.is_in_port_connected(port_idx):
-                    if port_idx != port_idx_to_connect:
-                        concat.add_input_port(port_idx_to_connect, skip_if_exist=True)
-                        concat.in_port(port_idx).get_connection().set_destination(concat.in_port(port_idx_to_connect))
-                    port_idx_to_connect += 1
-                elif port_idx in concat.in_ports():
-                    concat.delete_input_port(port_idx)
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/middle/ConcatOptimization.py:116]: why did you remove this transformation?

**Review by rkazants — CHANGES_REQUESTED** [review]
removed transformation that is still used. Please recover it

File: tools/mo/openvino/tools/mo/middle/ConcatOptimization.py (hunk 5 above)
Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/middle/ConcatOptimization.py:116]: Thanks for noticing! I removed by mistake `ConcatOptimization` is not used but `ConcatOdInputEraserAndPortsReconnect` is used. You are right. Recovered it

**Comment by pavel-esir** [comment]
> removed transformation that is still used. Please recover it

Thanks for noticing! Recovered the transformation

File: tools/mo/automation/package_BOM.txt [hunk 6]
```diff
@@ -716,7 +716,6 @@ openvino/tools/mo/middle/BiasAddBroadcasting.py
 openvino/tools/mo/middle/BinarizeWeightsM1P1.py
 openvino/tools/mo/middle/BlockLSTMtoLSTMSequence.py
 openvino/tools/mo/middle/CheckForCycle.py
-openvino/tools/mo/middle/ConcatOptimization.py
```
Comment (rkazants) [inline tools/mo/automation/package_BOM.txt:719]: Now you seem not to delete this

**Review by rkazants — CHANGES_REQUESTED** [review]
please double-check your PR

File: tools/mo/automation/package_BOM.txt (hunk 6 above)
Comment (pavel-esir) [reply tools/mo/automation/package_BOM.txt]: done

File: tools/mo/openvino/tools/mo/middle/fusings.py [hunk 7]
```diff
@@ -46,9 +46,6 @@ def find_and_replace_pattern(self, graph: Graph):
         for_graph_and_each_sub_graph_recursively(graph, fuse_pad)
         for_graph_and_each_sub_graph_recursively(graph, lambda G: G.clean_up())
 
-        # Mark nodes with attr 'can_be_fused': False to disable fusing for specified nodes
-        for_graph_and_each_sub_graph_recursively(graph, lambda graph: mark_unfused_nodes(graph, argv.finegrain_fusing))
-
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/middle/fusings.py:51]: why did we remove it?
@jane-intel, could you please comment on this change if it is correct?

File: tools/mo/unit_tests/moc_tf_fe/check_info_messages.py [hunk 8]
```diff
@@ -46,7 +46,6 @@ def arg_parse_helper(input_model,
         source_layout={},
         target_layout={},
         freeze_placeholder_with_value=None,
-        tensorflow_use_custom_operations_config=None,
```
Comment (rkazants) [inline tools/mo/unit_tests/moc_tf_fe/check_info_messages.py:49]: here we remove `tensorflow_use_custom_operations_config` option. Should we?

File: tools/mo/unit_tests/moc_tf_fe/check_info_messages.py [hunk 9]
```diff
@@ -46,7 +46,6 @@ def arg_parse_helper(input_model,
...
         data_type=None,
         tensorflow_custom_operations_config_update=None,
```
Comment (rkazants) [inline tools/mo/unit_tests/moc_tf_fe/check_info_messages.py:50]: What is our plan about this option `tensorflow_custom_operations_config_update` and is the difference?
@slyalin, @jane-intel, could you please comment on it?

File: tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py [hunk 10]
```diff
@@ -43,14 +43,8 @@ def base_args_config():
     args.scale_values = None
     args.output_dir = os.getcwd()
     args.freeze_placeholder_with_value = None
-    args.tensorflow_use_custom_operations_config = None
```
Comment (rkazants) [inline tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py:46]: here we remove tensorflow_use_custom_operations_config option. Should we, if we are not going to remove it in this PR?

File: tools/mo/openvino/tools/mo/middle/fusings.py (hunk 7 above)
Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/middle/fusings.py:51]: because this transformation is run only if `finegrain_fusing` is enabled, but since we removed this argument this marking pass is newer run I removed it

Comment (rkazants) [reply tools/mo/openvino/tools/mo/middle/fusings.py:51]: What is the default value of this option?

File: tools/mo/unit_tests/moc_tf_fe/check_info_messages.py (hunk 8 above)
Comment (pavel-esir) [reply tools/mo/unit_tests/moc_tf_fe/check_info_messages.py:49]: Specifying this argument was not obligatory for this unit-tests. Even though this arg will be removed from `cli_parser` in different PR let's remove this trashy unnecessary arg setting?

File: tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py (hunk 10 above)
Comment (pavel-esir) [reply tools/mo/unit_tests/moc_tf_fe/conversion_basic_models.py:46]: Answered above. Let's continue there

File: tools/mo/unit_tests/moc_tf_fe/check_info_messages.py (hunk 9 above)
Comment (rkazants) [reply tools/mo/unit_tests/moc_tf_fe/check_info_messages.py:50]: This does not block this PR to proceed

File: tools/mo/openvino/tools/mo/middle/fusings.py (hunk 7 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/middle/fusings.py:51]: Remained question. Please double-check if this is correct change. Don't we skip some fusing transformation this way?

Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/middle/fusings.py:51]: I double checked. Default value is `None` which means skip and do not mark anything which is equivalent of not running this pass 
image

**Review by rkazants — APPROVED** [review]
